w08h02/test/pgdp/pools/TaskFunctionBehaviorTest.java

```diff
@@ -143,4 +143,25 @@ void equalsTwoDifferentTaskFunctions() {
         assertFalse(task1.equals(task2), "equals should yield false for two different TaskFunction instances");
     }
 
+    @Test
+    @DisplayName("equals() should return false for identical input and manipulated subclassed function")
+    void equalsUnderlyingSubclassedFunction() {
+        var func1 = new TaskFunction<Integer, Integer>(SQUARE);
+        var func2 = new TaskFunction<Integer, Integer>(SQUARE) {
+            @Override
+            // In case TaskFunction::equals uses getID() which changes symmetry of equals
+            public int getID() {
+                return func1.getID();
+            }
+
+            @Override
+            // In case TaskFunction::equals uses hashCode() which changes symmetry of equals
+            public int hashCode() {
+                return func1.hashCode();
+            }
+        };
+
+        assertFalse(func1.equals(func2), "equals should yield false for task with overwritten methods");
+    }
```

Comment:
Are you sure that we should have getter for ID?

Reply:
They have not explicitly mentioned it, but it would make sense. I can leave the getID() there but extract the value by reflection. So we can still test for potential problems for people that have implemented the getter.

Reply:
That @Override for the getID must be removed anyway, in case that function does not exist. Getter/Setter are not required and cannot be expected. They are also not expected by public tests.

Reply:
> That @OverRide for the getID must be removed anyway, in case that function does not exist. Getter/Setter are not required and cannot be expected. They are also not expected by public tests. Generally, I think those tests are really useful!

Thanks for pointing that out.